pages/editPetPage.ts

```diff
@@ -0,0 +1,42 @@
+import { Page, expect } from "@playwright/test"
+import { HelperBase } from "./helperBase"
+
+export class EditPetPage extends HelperBase{ 
+    
+    constructor(page : Page){
+        super(page)
+    }
+
+    async validateNameOwnerAndPetTypeFieldsDisplayCorrectly(petName: string, owner: string, petType: string){
+        // Validate that the name input field is correct
+        await expect(this.page.locator('#name')).toHaveValue(petName)
+        await expect(this.page.locator('#owner_name')).toHaveValue(owner)
+        await expect(this.page.locator('#type1')).toHaveValue(petType)
+
+    }
+
+    async validateAllPetValuesFromDropdownDisplayedInPetTypeField(){
+
+        // Iterate through each pet type option to validate that the pet is correctly listed in the textbox to the left 
+        const allPetTypeOptions = await this.page.locator("option").allTextContents()
```

Comment:
This method can be potentially false postitve.
If `allPetTypeOptions` will be empty, the for loop below will not be executed so assertion simply will be skipped.
Currently this method is working because test case has a previous step with assertions `validateNameOwnerAndPetTypeFieldsDisplayCorrectly()` and because of that, page is properly loaded. But if you remove this step from the test case, you will see how the loop not executed.

Reply:
Good catch!  I'm thinking about this one... I'm going to ask my question in Slack.